src/screens/LoadingAddresses.js

```diff
@@ -61,9 +61,15 @@ class LoadingAddresses extends React.Component {
     }
     // To prevent only a blink in this screen when user loads the addresses really fast
     // I set that the user will see this screen at least for 2 seconds
-    setTimeout(() => {
+    const timeoutId = setTimeout(() => {
       this.setState({ canRedirect: true });
     }, 2000);
+    this.setState({ timeoutId });
+  }
+
+  componentWillUnmount = () => {
+    clearTimeout(this.state.timeoutId);
```

Comment:
Is there any chance we might get here before setting `timeoutId`? I would add a `if (this.timeoutId)` just to guarantee we don't get any errors here.

Reply:
`componentWillUnmount` will never run before `componentDidMount`, but if the `setTimeout` callback has already been called this `clearTimeout` will be a no-op.